src/assets/wise5/services/projectTagService.ts

```diff
@@ -38,10 +38,12 @@ export class ProjectTagService {
     });
   }
 
-  createTag(tagName: string): Subscription {
-    return this.http.post(`/api/user/tag`, { text: tagName }).subscribe((tag: Tag) => {
-      this.newTagSource.next(tag);
-    });
+  createTag(tagName: string): Observable<Object> {
+    return this.http.post(`/api/user/tag`, { text: tagName });
+  }
+
+  emitNewTag(tag: Tag): void {
+    this.newTagSource.next(tag);
   }
```

Comment:
I'm guessing that we always want to emit the newTagSource event after each call here. If you pipe the result to post(), you can emit the newTagSource event and not need to define emitNewTag() function?
```suggestion
  createTag(tagName: string): Observable<Tag> {
    return this.http
      .post<Tag>(`/api/user/tag`, { text: tagName })
      .pipe(
        tap((tag) => {
          this.newTagSource.next(tag);
        })
      );
  }
```